floodviz/static/js/map.js

```diff
@@ -0,0 +1,106 @@
+document.addEventListener("DOMContentLoaded", function (event) {
+    "use strict";
+
+    // Define helper functions
+    function degreesToRadians(degrees) {
+        return degrees * Math.PI / 180;
+    }
+
+    function radiansToDegrees(radians) {
+        return radians * 180 / Math.PI;
+    }
+
+    // read in projection information
+    var proj = proj4(FV.mapinfo.proj4string);
+
+    var project = function (lambda, phi) {
+        return proj.forward([lambda, phi].map(radiansToDegrees));
+    };
+
+    project.invert = function (x, y) {
+        return proj.inverse([x, y]).map(degreesToRadians);
+    };
+
+    //Define map projection
+    var projection = d3.geoProjection(project);
+
+    // Give projection initial rotation and scale
+    projection
+        .scale(1)
+        .translate([0, 0]);
+
+    //Define path generator
+    var path = d3.geoPath().projection(projection);
+
+    var height = FV.mapinfo.height;
+    var width = FV.mapinfo.width;
+
+    //Create SVG element
+    var svg = d3.select("#map")
+        .append("svg")
+        .attr("width", width)
+        .attr("height", height);
+
+
+    // set bounding box to values provided
+    var b = path.bounds(FV.mapinfo.bounds);
+
+
+    var s = FV.mapinfo.scale / Math.max((b[1][0] - b[0][0]) / width, (b[1][1] - b[0][1]) / height);
+    var t = [(width - s * (b[1][0] + b[0][0])) / 2, (height - s * (b[1][1] + b[0][1])) / 2];
+
+    // Update the projection
+    projection
+        .scale(s)
+        .translate(t);
+
+
+    // add layers in sensible order
+    add_paths(FV.mapinfo.bg_data, "background");
+
+    add_paths(FV.mapinfo.rivers_data, "river");
+
+    add_circles(FV.mapinfo.ref_data, "ref-point", 2);
+
+    add_circles(FV.mapinfo.site_data, "gage-point", 3);
+
+
+    if (FV.mapinfo.debug) {
```

Comment:
I don't see that debug is getting set in mapinfo. Consider putting the config DEBUG value in FV. So in index.html it would be FV = {debug: config.DEBUG}